File: android/src/main/kotlin/com/gstory/flutter_tencentad/FlutterTencentAdViewPlugin.kt
package com.gstory.flutter_tencentad

import android.app.Activity
import com.gstory.flutter_tencentad.bannerad.BannerAdViewFactory
import com.gstory.flutter_tencentad.expressad.NativeExpressAdViewFactory
import com.gstory.flutter_tencentad.splashad.SplashAdViewFactory
import io.flutter.embedding.engine.plugins.FlutterPlugin

/**
 * @Description:
 * @Author: gstory0404@gmail
 * @CreateDate: 2021/8/7 11:38
 */

object FlutterTencentAdViewPlugin {
    fun registerWith(binding: FlutterPlugin.FlutterPluginBinding, activity: Activity) {
        //注册banner广告
        binding.platformViewRegistry.registerViewFactory(FlutterTencentAdConfig.bannerAdView, BannerAdViewFactory(binding.binaryMessenger, activity))
        //注册splash广告
        binding.platformViewRegistry.registerViewFactory(FlutterTencentAdConfig.splashAdView, SplashAdViewFactory(binding.binaryMessenger,activity))
        //注册Express广告
        binding.platformViewRegistry.registerViewFactory(FlutterTencentAdConfig.nativeExpressAdView, NativeExpressAdViewFactory(binding.binaryMessenger,activity))
    }
}

File: ios/Classes/FlutterTencentadViewPlugin.swift
//
//  FlutterTencentadViewPlugin.swift
//  flutter_tencentad
//
//  Created by gstory on 2021/9/1.
//

import Foundation
import Flutter

class FlutterTencentadViewPlugin{
    static func register(viewRegistrar : FlutterPluginRegistrar){
       //banner广告
        let bannerAdViewFactory = BannerAdViewFactory(messenger: viewRegistrar.messenger())
        viewRegistrar.register(bannerAdViewFactory, withId: FlutterTencentadConfig.view.bannerAdView)
        //spalsh广告
        let splashAdViewFactory = SplashAdViewFactory(messenger: viewRegistrar.messenger())
        viewRegistrar.register(splashAdViewFactory, withId: FlutterTencentadConfig.view.splashAdView)
        //信息流广告
        let nativeAdviewFactory = NativeExpressAdViewFactory(messenger: viewRegistrar.messenger())
        viewRegistrar.register(nativeAdviewFactory, withId: FlutterTencentadConfig.view.nativeAdView)
    }
}
